miden-lib/asm/sat/layout.masm

```diff
@@ -0,0 +1,396 @@
+    exec.constants::get_note_mem_size mul push.CREATED_NOTE_SECTION_OFFSET add
+end
+
+#! Returns the created note recipient
+#!
+#! Stack: [created_note_data_ptr]
+#! Output: [R]
+#!
+#! - created_note_data_ptr is the memory address at which the created note data begins.
+#! - R is the recipient of the created note.
+export.get_created_note_recipient
+    padw
+    movup.4 push.CREATED_NOTE_RECIPIENT_OFFSET add
+    mem_loadw
+end
+
+#! Returns the number of assets in the created note
+#!
+#! Stack: [created_note_data_ptr]
+#! Output: [num_assets]
+#!
+#! - created_note_data_ptr is the memory address at which the created note data begins.
+#! - num_assets is the number of assets in the created note.
+export.get_created_note_num_assets
+    push.CREATED_NOTE_NUM_ASSETS_OFFSET add mem_load
+end
+
+#! Returns a pointer to the created note asset data
+#!
+#! Stack: [created_note_data_ptr]
+#! Output: [asset_data_ptr]
+#!
+#! - created_note_data_ptr is the memory address at which the created note data begins.
+#! - asset_data_ptr is the memory address at which the created note asset data begins.
+export.get_created_note_asset_data_ptr
+    push.CREATED_NOTE_ASSETS_OFFSET add
+end
+
+#! Sets the created note vault hash
+#!
+#! Stack: [created_note_data_ptr, V]
+#! Output: []
+#!
+#! - created_note_data_ptr is the memory address at which the created note data begins.
+#! - V is the vault hash of the created note.
+export.set_created_note_vault_hash
+    push.CREATED_NOTE_VAULT_HASH_OFFSET add mem_storew
+end
+
+# CONSUMED NOTES
+# -------------------------------------------------------------------------------------------------
+
+#! Sets the number of consumed notes.
+#!
+#! Stack: [num_consumed_notes]
+#! Output: []
+#!
+#! - num_consumed_notes is the number of consumed notes.
+export.set_num_consumed_notes
+    push.CONSUMED_NOTE_NUM_PTR mem_store
+end
+
+#! Computes a pointer to the memory address at which the data associated with a consumed note with
+#! index i is stored.
+#!
+#! Stack: [i]
+#! Output: [ptr]
+#!
+#! - i is the index of the consumed note.
+#! - ptr is the memory address of the data segment for consumed note i.
+export.get_consumed_note_ptr
+    add.1 exec.constants::get_note_mem_size mul push.CONSUMED_NOTE_SECTION_OFFSET add
+end
+
+#! Computes a pointer to the memory address at which the nullifier associated a note with index i
+#! is stored.
+#!
+#! Stack: [i]
+#! Output: [ptr]
+#!
+#! - i is the index of the consumed note.
+#! - ptr is the nullifier memory address for note i.
+export.get_consumed_note_nullifier_ptr
+    push.CONSUMED_NOTE_SECTION_OFFSET.1 add add
+end
+
+#! Returns the consumed note nullifier
+#!
+#! Stack: [i]
+#! Output: [nullifier]
+#!
+#! - i is the index of the consumed note.
+#! - nullifier is the nullifier of the consumed note.
+export.get_consumed_note_nullifier
+    padw movup.4 push.CONSUMED_NOTE_SECTION_OFFSET.1 add add mem_loadw
+end
+
+#! Returns the consumed note script root
```

Comment:
I'd probably clarify this a bit more - maybe something like this: "Returns the script root of a consumed note located at the specified memory address."